tests/test_merge_db_functions.py

```diff
@@ -0,0 +1,477 @@
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+###############################################################################
+import sys
+import os
+import sqlite3
+
+from tuna.utils.merge_db import parse_jobline, parse_text_fdb_name, parse_text_pdb_name, load_master_list
+from tuna.utils.merge_db import best_solver, target_merge, no_job_merge, single_job_merge
+from tuna.utils.merge_db import multi_job_merge, update_master_list, write_merge_results
+from tuna.utils.merge_db import merge_text_file, merge_sqlite_pdb, merge_sqlite_bin_cache
+from tuna.utils.merge_db import merge_sqlite, get_file_list, get_sqlite_table
+from tuna.utils.merge_db import get_sqlite_row, get_sqlite_data
+from tuna.helper import prune_cfg_dims
+
+
+def test_parse_jobline():
+
+  data = """1-160-698-5x5-64-79-348-1-1x1-2x2-1x1-0-NCHW-FP32-F=miopenConvolutionFwdAlgoImplicitGEMM:ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used;miopenConvolutionFwdAlgoWinograd:ConvBinWinogradRxSf2x3g1,0.03856,0,miopenConvolutionFwdAlgoWinograd,not used;miopenConvolutionFwdAlgoDirect:ConvOclDirectFwdGen,0.0536,0,miopenConvolutionFwdAlgoDirect,not used;miopenConvolutionFwdAlgoGEMM:GemmFwdRest,0.05712,2749200,miopenConvolutionFwdAlgoGEMM,not used\n"""
+
+  key, vals = parse_jobline(data)
+
+  assert (key == '1-160-698-5x5-64-79-348-1-1x1-2x2-1x1-0-NCHW-FP32-F')
+  assert (vals == {
+      'miopenConvolutionFwdAlgoImplicitGEMM':
+          'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used',
+      'miopenConvolutionFwdAlgoWinograd':
+          'ConvBinWinogradRxSf2x3g1,0.03856,0,miopenConvolutionFwdAlgoWinograd,not used',
+      'miopenConvolutionFwdAlgoDirect':
+          'ConvOclDirectFwdGen,0.0536,0,miopenConvolutionFwdAlgoDirect,not used',
+      'miopenConvolutionFwdAlgoGEMM':
+          'GemmFwdRest,0.05712,2749200,miopenConvolutionFwdAlgoGEMM,not used'
+  })
+
+  algo = 'miopenConvolutionFwdAlgoImplicitGEMM'
+  assert (
+      vals[algo] ==
+      'ConvHipImplicitGemmForwardV4R4Xdlops_Padded_Gemm,0.02352,0,miopenConvolutionFwdAlgoImplicitGEMM,not used'
+  )
+
+
+def test_parse_text_fdb_name():
+  master_file = '/data/tests/gfx90a_68.HIP.fdb.txt'
+
+  arch, num_cu, final_file, copy_files = parse_text_fdb_name(master_file)
+  assert (arch == "gfx90a")
+  assert (num_cu == '68')
+  assert (final_file == "/data/tests/gfx90a_68.HIP.fdb.txt")
+  assert (copy_files == [])
+
+
+def test_parse_text_pdb_name():
+  master_file = '/data/test/gfx1030_36.cd.pdb.txt'
```

Comment:
This doesn't look like a typical pdb filename 'gfx1030_36.db' is typical.

Reply:
added new test case to check the pdb num_cu check.